docs/how-to/data/crictl/busybox/container_config.json

```diff
@@ -0,0 +1,18 @@
+{
+	"metadata": {
+		"name": "busybox-container"
```

Comment:
Hi @liubin - I'm having trouble testing these:

```bash
$ cat /etc/crio/crio.conf.d/99-kata.conf
[crio.runtime]
default_runtime = "kata"

[crio.runtime.runtimes.kata]
runtime_path = "/opt/kata/bin/containerd-shim-kata-v2"
runtime_root = "/run/vc"
runtime_type = "vm"
privileged_without_host_devices = true

$ crio --version
crio version 1.18.3
Version:       1.18.3
GitCommit:     61de18161fb4ccda720768c001713592b5a04e46
GitTreeState:  clean
BuildDate:     2020-10-08T01:53:51Z
GoVersion:     go1.15.0
Compiler:      gc
Platform:      linux/amd64
Linkmode:      dynamic

$ crictl --version
crictl version v1.18.0

$ sudo /usr/local/bin/crictl --debug runp -r kata sandbox_config.json
DEBU[0000] RunPodSandboxRequest: &RunPodSandboxRequest{Config:&PodSandboxConfig{Metadata:&PodSandboxMetadata{Name:busybox-pod,Uid:busybox-pod,Namespace:,Attempt:0,},Hostname:busybox_host,LogDirectory:,DnsConfig:&DNSConfig{Servers:[],Searches:[],Options:[],},PortMappings:[]*PortMapping{},Labels:map[string]string{},Annotations:map[string]string{},Linux:&LinuxPodSandboxConfig{CgroupParent:,SecurityContext:nil,Sysctls:map[string]string{},},},RuntimeHandler:kata,}
DEBU[0000] RunPodSandboxResponse: nil
FATA[0000] run pod sandbox failed: rpc error: code = Unknown desc = setting pod sandbox name and id: cannot generate pod name without namespace
$ 
```

If I add a namespace, I get:

```bash
$ sudo /usr/local/bin/crictl --debug runp -r kata sandbox_config.json
DEBU[0000] RunPodSandboxRequest: &RunPodSandboxRequest{Config:&PodSandboxConfig{Metadata:&PodSandboxMetadata{Name:busybox-pod,Uid:busybox-pod,Namespace:foo.bar.baz,Attempt:0,},Hostname:busybox_host,LogDirectory:,DnsConfig:&DNSConfig{Servers:[],Searches:[],Options:[],},PortMappings:[]*PortMapping{},Labels:map[string]string{},Annotations:map[string]string{},Linux:&LinuxPodSandboxConfig{CgroupParent:,SecurityContext:nil,Sysctls:map[string]string{},},},RuntimeHandler:kata,}
DEBU[0001] RunPodSandboxResponse: nil
FATA[0001] run pod sandbox failed: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_busybox-pod_foo.bar.baz_busybox-pod_0(4efa96c415fe4a3239e8b1bf5e9c5ac04f57ac9454196a91d5662e32051d631c): error adding loopback interface: failed to Statfs "": no such file or directory
```

I think as a minimum we need to specify a namespace in the json files:

```diff
diff --git a/docs/how-to/data/crictl/busybox/container_config.json b/docs/how-to/data/crictl/busybox/container_config.json
index 57724965..fcea05a7 100644
--- a/docs/how-to/data/crictl/busybox/container_config.json
+++ b/docs/how-to/data/crictl/busybox/container_config.json
@@ -1,6 +1,7 @@
 {
        "metadata": {
-               "name": "busybox-container"
+               "name": "busybox-container",
+               "namespace": "test.kata"
        },
        "image": {
                "image": "docker.io/library/busybox:latest"
diff --git a/docs/how-to/data/crictl/busybox/sandbox_config.json b/docs/how-to/data/crictl/busybox/sandbox_config.json
index 963db863..d722555f 100644
--- a/docs/how-to/data/crictl/busybox/sandbox_config.json
+++ b/docs/how-to/data/crictl/busybox/sandbox_config.json
@@ -1,7 +1,8 @@
 {
        "metadata": {
                "name": "busybox-pod",
-               "uid": "busybox-pod"
+               "uid": "busybox-pod",
+               "namespace": "test.kata"
        },
        "hostname": "busybox_host",
        "log_directory": "",
```

The second error I get is probably due to me not having setup cni, but again, I wonder if there is a way to run without a network just for busybox as a simple demo?

/cc @chavafg, @GabyCT.

Reply:
Thanks @jodh-intel - @liubin PR works for me, I don't need to change anything

Reply:
@devimc pointed out that I'm hitting https://github.com/cri-o/cri-o/issues/3618. I'll re-test with https://github.com/cri-o/cri-o/releases/tag/v1.19.0 ...

Reply:
I see the same problem with crio + crictl v1.19.0:

```bash
$ /usr/local/bin/crictl --version
crictl version v1.19.0

$ crio --version
crio version 1.19.0
Version:       1.19.0
GitCommit:     99c925bebdd9e392f2d575e25f2e6a1082e6c232
GitTreeState:  clean
BuildDate:     2020-10-05T17:07:23Z
GoVersion:     go1.13.15
Compiler:      gc
Platform:      linux/amd64
Linkmode:      dynamic

$ sudo /usr/local/bin/crictl --debug runp -r kata sandbox_config.json
DEBU[0000] get runtime connection
DEBU[0000] connect using endpoint 'unix:///var/run/crio/crio.sock' with '2s' timeout
DEBU[0000] connected successfully using endpoint: unix:///var/run/crio/crio.sock
DEBU[0000] RunPodSandboxRequest: &RunPodSandboxRequest{Config:&PodSandboxConfig{Metadata:&PodSandboxMetadata{Name:busybox-pod,Uid:busybox-pod,Namespace:,Attempt:0,},Hostname:busybox_host,LogDirectory:,DnsConfig:&DNSConfig{Servers:[],Searches:[],Options:[],},PortMappings:[]*PortMapping{},Labels:map[string]string{},Annotations:map[string]string{},Linux:&LinuxPodSandboxConfig{CgroupParent:,SecurityContext:nil,Sysctls:map[string]string{},},},RuntimeHandler:kata,}
DEBU[0000] RunPodSandboxResponse: nil
FATA[0000] run pod sandbox: rpc error: code = Unknown desc = setting pod sandbox name and id: cannot generate pod name without namespace

Reply:
@jodh-intel Could you retry with containerd? Just so that we can narrow out any issue on kata side.

Reply:
@jodh-intel It seems that CRI-O need `namespace` field and I added it. For network issue, I can't find a way to skip the network settings yet :-)